consensus/service/src/test_utils/mod.rs

```diff
@@ -25,17 +24,12 @@ impl UntrustedInterfaces for TrivialTxManagerUntrustedInterfaces {
         Ok(())
     }
 
-    fn combine(
-        &self,
-        tx_contexts: &[&WellFormedTxContext],
-        max_elements: usize,
-    ) -> BTreeSet<TxHash> {
-        BTreeSet::from_iter(
-            tx_contexts
-                .iter()
-                .take(max_elements)
-                .map(|tx_context| tx_context.tx_hash())
-                .cloned(),
-        )
+    fn combine(&self, tx_contexts: &[&WellFormedTxContext], max_elements: usize) -> Vec<TxHash> {
+        tx_contexts
```

Comment:
The previous implementation ensured the returned set of values is sorted (by TxHash).
This implementation does not sort values, which I believe is not a correct implementation of `combine`.

Reply:
Good catch!